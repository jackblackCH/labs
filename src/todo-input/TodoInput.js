import React from 'react';
import './todo-input.css';

const TodoInput = (props) => <input className="c-todo-input" onChange={props.onChange} value={props.value} />

export default TodoInput;
