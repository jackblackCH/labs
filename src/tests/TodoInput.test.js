import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TodoInput from '../todo-input/TodoInput';

it('should render an input', () => {
	const tree = toJson(shallow(<TodoInput value="hello" />));
	expect(tree).toMatchSnapshot();
});

it('should call props.onChange when the input changes', () => {
	const onChange = jest.fn();
	const component = shallow(<TodoInput value="hello" onChange={onChange} />);
	const evt = { target: { value: 'bla' } };
	component.find('input').simulate('change', evt)
	expect(onChange).toBeCalledWith(evt);
});
