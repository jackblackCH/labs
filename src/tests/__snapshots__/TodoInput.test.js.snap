exports[`test should render an input 1`] = `
<input
  className="c-todo-input"
  value="hello" />
`;
